extensions/ql-vscode/src/legacy-query-server/run-queries.ts

```diff
@@ -327,17 +335,15 @@ export async function compileAndRunQueryAgainstDatabase(
   // won't trigger this check)
   // This test will produce confusing results if we ever change the name of the database schema files.
   const querySchemaName = basename(packConfig.dbscheme);
-  const dbSchemaName = basename(dbItem.contents.dbSchemeUri.fsPath);
+  const dbSchemaName = basename(dbContents.dbSchemeUri?.fsPath);
   if (querySchemaName !== dbSchemaName) {
     void extLogger.log(
       `Query schema was ${querySchemaName}, but database schema was ${dbSchemaName}.`,
     );
     throw new Error(
       `The query ${basename(
-        initialInfo.queryPath,
-      )} cannot be run against the selected database (${
-        dbItem.name
-      }): their target languages are different. Please select a different database and try again.`,
+        query.queryPath,
+      )} cannot be run against the selected database: their target languages are different. Please select a different database and try again.`,
```

Comment:
Minor: Would be nice to continue using the database name here, but not sure if that value is available with this refactoring.

Reply:
If this were the new query server, I probably would have put more effort into that. Since it's just for the legacy server, it didn't seem worthwhile.